test/testuninitvar.cpp

```diff
@@ -3175,6 +3175,21 @@ class TestUninitVar : public TestFixture {
                        "}\n", "test.c");
         ASSERT_EQUALS("", errout.str());
 
+        checkUninitVar("struct AB { int a; int b; };\n"  // pass struct members to memcmp by address
+                       "void f(void) {\n"
+                       "    struct AB ab;\n"
+                       "    (void)memcmp(&ab.a, &ab.b, sizeof(int));\n"
```

Comment:
I guess you use the wrong function in these tests, you probably want to use a function that has "out" parameters.

Reply:
I do not see any easy direct test for this. As far as I see, the function is designed to be "safe". If uncertain it will assume that the variable is written.

One way would be to add a function parameter "unsafe" that is false by default. And in the tests call `isMemberVariableAssigned` directly and try to set that to true.

Reply:
> I guess you use the wrong function in these tests, you probably want to use a function that has "out" parameters.

I wanted to use a function that uses "in" values because then the variables are still uninitialized after the function call which should be detected by Cppcheck. But since every "in" argument is normally also configured with `<not-uninit/>` this does not really make sense.

Reply:
> I do not see any easy direct test for this. As far as I see, the function is designed to be "safe". If uncertain it will assume that the variable is written.
> 
> One way would be to add a function parameter "unsafe" that is false by default. And in the tests call `isMemberVariableAssigned` directly and try to set that to true.

Hmm, i do not like to alter the function only to make it possible that it can be tested directly. IMHO that makes it more complex than necessary and more error prone.

Maybe a piece of library configuration can be loaded where only the argument direction is set (like in testlibrary). And the test uses this library instead of a function declaration. But i am not sure if this is possible without greater effort.

Reply:
> Hmm, i do not like to alter the function only to make it possible that it can be tested directly. IMHO that makes it more complex than necessary and more error prone.

ok

> And the test uses this library instead of a function declaration. But i am not sure if this is possible without greater effort.

If you would like to do that then I think that TestLibrary shows how you can load a custom configuration. I'd suggest you configure a special function `dostuff` or something that is not configured in our `std.cfg`.

Reply:
I have done that and committed changed tests.